addon/models/learner-group.js

```diff
@@ -114,22 +114,30 @@ export default DS.Model.extend({
     });
   }),
 
-  allDescendantUsers: computed('users.[]', 'children.@each.users', function(){
-    var deferred = Ember.RSVP.defer();
-    this.get('users').then(users => {
-      this.get('children').then(children => {
-        map(children.mapBy('allDescendantUsers'), childUsers => {
-          users.addObjects(childUsers);
-        }).then(() => {
-          deferred.resolve(users.uniq());
-        });
-      });
-    });
+  /**
+   * A list of all users in this group and any of its sub-groups.
+   * @property allDescendantUsers
+   * @type {Ember.computed}
+   * @public
+   */
+  allDescendantUsers: computed('users.[]', 'children.@each.allDescendantUsers', async function(){
+    const allUsers = [];
+    let users = await this.get('users').toArray();
+    let subgroups = await this.get('children').toArray();
 
-    return DS.PromiseArray.create({
-      promise: deferred.promise
+    let usersInSubgroups = await map(subgroups.mapBy('allDescendantUsers'), allDescendantUsers => {
+      return allDescendantUsers;
     });
+
+    allUsers.pushObjects(usersInSubgroups.reduce((array, set) => {
```

Comment:
Pushing objects onto an otherwise empty array seems overly complicated. Why not just:

```
let allUsers = usersInSubgroups.reduce((array, set) => {
...
});
```